### PR TITLE
Add Transaction.signingData

### PR DESCRIPTION
**Pull request by sublimator**
This is needed for ed25519, which, per the rippled implementation, uses its own hashing routine (sha512) and wants the full data. 

A small milestone pull request ...

### CODE REVIEW

File: src/core/transaction.js [hunk 1]
```diff
@@ -435,6 +435,15 @@ Transaction.prototype.signingHash = function(testnet) {
   return this.hash(testnet ? 'HASH_TX_SIGN_TESTNET' : 'HASH_TX_SIGN');
 };
 
+Transaction.prototype.signingData = function() {
+  return this._signingData('HASH_TX_SIGN').buffer;
+};
+
+Transaction.prototype._signingData = function(prefix) {
```
Comment (sublimator) [inline src/core/transaction.js]: Kinda pointless taking the prefix, given test net functionality is removed from rippled ...

File: src/core/transaction.js [hunk 2]
```diff
@@ -435,6 +435,15 @@ Transaction.prototype.signingHash = function(testnet) {
...
+  return SerializedObject.from_json(
+        this.tx_json, hashprefixes[prefix].toString(16));
```
Comment (sublimator) [inline src/core/transaction.js]: I guess theoretically there could be a problem if the hash prefix started with a 0.

**Comment by wltsmrz** [comment]
LGTM

File: bin/ci.sh [hunk 3]
```diff
@@ -10,11 +10,9 @@ typecheck() {
 
 lint() {
   REPO_URL="https://raw.githubusercontent.com/ripple/javascript-style-guide"
-  npm install -g eslint
```
Comment (clark800) [inline bin/ci.sh]: Good idea to avoid double installing

**Pull request merged**